File: leetcode/697.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findShortestSubArray = function (nums) {
  let hash = {};
  for (let i = 0; i < nums.length; i++) {
    let num = nums[i];
    if (!hash[num]) hash[num] = [];
    hash[num].push(i);
  }
  let maxFreq = 0;
  let minLen = Infinity;
  for (let key of Object.keys(hash)) {
    if (hash[key].length > maxFreq) {
      maxFreq = hash[key].length;
      minLen = hash[key][maxFreq - 1] - hash[key][0] + 1;
    } else if (hash[key].length === maxFreq) {
      minLen = Math.min(hash[key][maxFreq - 1] - hash[key][0] + 1, minLen);
    }
  }
  return minLen;
};

File: leetcode/697.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        freq = dict()
        max_freq = 0
        min_len = len(nums)
        for i, n in enumerate(nums):
            if n not in freq:
                freq[n] = []
            freq[n].append(i)
            max_freq = max(max_freq, len(freq[n]))

        ans = len(nums)
        for k in freq:
            if len(freq[k]) == max_freq:
                ans = min(ans, freq[k][-1] - freq[k][0] + 1)
        return ans
